src/tiled/objectselectionitem.cpp

```diff
@@ -199,16 +199,21 @@ void MapObjectLabel::syncWithMapObject(const MapRenderer &renderer)
     QPointF screenPos = renderer.pixelToScreenCoords(mObject->position());
     QRectF bounds = mObject->screenBounds(renderer);
 
-    // Adjust the bounding box for object rotation
-    bounds = rotateAt(screenPos, mObject->rotation()).mapRect(bounds);
+    if (mObject->shape() == MapObject::Point) {
+        // Use screenspace offset, since the Point annotation rescales with the zoom-level
+        setPos(screenPos - QPointF{ 0, bounds.height() / renderer.painterScale()});
```

Comment:
The problem is that the position set with `setPos` will be scaled with the view, even if this item will not. Previously this is what we wanted, but now that we don't, we should not use `setPos` to center the label above the object but rather apply an offset when rendering the label (and to the bounding rect of this item). Of course, this should still only be done for point objects, since for all other objects we do want the position of the label to be affected by the scaling.

Reply:
@bjorn how does this look now?  In my other branch I used a similar approach but didn't realize that the repaint was misbehaving because the bounding box was not also translated.